### PR TITLE
Remove cashay queries

### PR DESCRIPTION
**Pull request by mattkrick**
this finishes removing cashay from the app

tests
- [ ] can log in
- [ ] segment gets notified about your identity & page events (confirm on segment.io >> development server)
- [ ] can change your picture
- [ ] can change your name
- [ ] can log out
- [ ] can change your org's picture

### CODE REVIEW

**Comment by mattkrick** [comment]
@dan-f ready to merge

**Comment by dan-f** [comment]
Manual tests working on the dev server! (Not sure how to get into our account on segment, will ask in slack).

- [x] can log in
- [ ] segment gets notified about your identity & page events (confirm on segment.io >> development server)
- [x] can change your picture
- [x] can change your name
- [x] can log out
- [x] can change your org's picture

On to the code review!

**Comment by dan-f** [comment]
BTW, I'm seeing a lot of query modules that have been removed. I realize some of them have been split up and moved into other modules, while some of them have been simply removed. Could you summarize the changes to the schema?

**Comment by mattkrick** [comment]
most of them haven't been used since 8,5, i'm just cleaning up the loose ends now. There are some queries that we had to use for cashay + graphql before graphql supported subscriptions. For example, we had to get the initial count of how many team members there were because team members would trickle in via socket. Now queries & subs are broken up so we don't have that problem.

the rest is just refactoring, breaking each into its own file instead of 1 file with many entities.

i also renamed `updateUserWithAuthToken` to `login`.

File: src/universal/mutations/UpdateUserProfileMutation.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      preferredName
+      picture
+    }
+  }
+`;
+
+const mutation = graphql`
+  mutation UpdateUserProfileMutation($updatedUser: UpdateUserProfileInput!) {
+    updateUserProfile(updatedUser: $updatedUser) {
+      ...UpdateUserProfileMutation_notification @relay(mask: false)
```
Comment (dan-f) [inline src/universal/mutations/UpdateUserProfileMutation.js]: Help me understand why:

- ... we're declaring fragments in this mutation module that are then getting imported in a subscription module
- ... we're masking these fragments on the mutation? Presumably we don't have fragment containers that are querying data directly off of a mutation response.

File: src/universal/mutations/fragments/UserAnalyticsFrag.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+graphql`
```
Comment (dan-f) [inline src/universal/mutations/fragments/UserAnalyticsFrag.js]: This is an interesting pattern; declaring fragments in their own modules. Since this fragment is just on the `User` type, I don't see why it should have to live under the `mutations` directory in particular.

File: src/universal/redux/AuthEngine.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ export default class AuthEngine {
     this.reducerName = reducerName;
   }
   saveToken(name, token, options, callback) {
```
Comment (dan-f) [inline src/universal/redux/AuthEngine.js]: `options`?

File: src/universal/redux/AuthEngine.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ export default class AuthEngine {
...
-    this.store.dispatch(setAuthToken(token, this.reducerName));
+    this.store.dispatch(setAuthToken(token));
     if (callback) callback(null, token);
```
Comment (dan-f) [inline src/universal/redux/AuthEngine.js]: Not sure why we need a callback if this is synchronous and we're just calling back with the same value that was provided to this function as a parameter.

File: src/universal/redux/__tests__/AuthEngine.test.js [hunk 5]
```diff
@@ -31,7 +31,6 @@ test('saveToken saves token', (done) => {
   const cb = (err, token) => {
     expect(err).toBe(null);
     expect(token).toBe(testToken);
-    expect(segmentActions.segmentEventIdentify).toBeCalled();
```
Comment (dan-f) [inline src/universal/redux/__tests__/AuthEngine.test.js]: Yet there still is a case where we _do_ want the `segmentEventIdentify` method to be called, so let's test it!

File: src/universal/redux/makeReducer.js [hunk 6]
```diff
@@ -34,7 +33,6 @@ const formPlugin = formPluginFactory();
 
 const appReducers = {
   [DEFAULT_AUTH_REDUCER_NAME]: auth,
-  cashay: cashayReducer,
```
Comment (dan-f) [inline src/universal/redux/makeReducer.js]: 👏

File: e2e/tests/authentication.test.js [hunk 7]
```diff
@@ -141,30 +141,30 @@ describe('Authentication', () => {
...
+  //it('can log in (and out) with valid credentials', async () => {
+  //  const { credentials } = cache;
+  //  expect(credentials).toBeTruthy();
+  //  await user.goToHomepage();
+  //  await user.openLoginModal();
+  //  await user.login(credentials);
+  //  await user.shouldSeeWelcomeWizard();
+  //  await user.logout();
+  //  await user.shouldSeeHomepage();
+  //});
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply e2e/tests/authentication.test.js]: for sure! i just tried & failed for an hour & gave up. CLI envs aren't my jam.

File: src/server/__tests__/utils/signupTeam.js [hunk 8]
```diff
@@ -1,60 +0,0 @@
-import mintToken, {mintTokenSigned} from './mintToken';
-
-import invitationMutation from '../../graphql/models/Invitation/invitationMutation';
-import teamMemberMutation from '../../graphql/models/TeamMember/teamMemberMutation';
-import teamMutation from '../../graphql/models/Team/teamMutation';
-import userMutation from '../../graphql/models/User/userMutation';
-
-export default function signupTeam(team, teamLeader, teamMembers, refreshAuthToken) {
-  describe('signup team', () => {
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/__tests__/utils/signupTeam.js]: oh this test is from waaaaay back in the day! it was tests like this that drove me to MockDB... the prep work necessary was painful (and like 50x slower iirc).

File: src/server/graphql/mutations/createFirstTeam.js [hunk 9]
```diff
@@ -47,16 +47,18 @@ export default {
...
     });
-    sendSegmentEvent('Welcome Step2 Completed', userId, {teamId});
+    sendSegmentEvent('Welcome Step2 Completed', viewerId, {teamId});
+
     return {
       team,
       teamLead,
-      jwt: tmsSignToken(authToken, tms)
+      jwt: tmsSignToken(authToken, tms),
+      userId: viewerId
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/mutations/createFirstTeam.js]: oh this doesn't make its way to the client, it's just used to generate the mutation payload. we could grab it from the auth token in CreateFirstTeamPayload, but then if we decide to publish to a pusub (ie keep their 2nd tab up to date), we'd need to revert back.

File: src/server/graphql/mutations/createOrgPicturePutUrl.js [hunk 10]
```diff
@@ -33,7 +33,7 @@ const createOrgPicturePutUrl = {
 
     // RESOLUTION
     const partialPath = `Organization/${orgId}/picture/${shortid.generate()}.${ext}`;
-    const url = getS3PutUrl(contentType, contentLength, partialPath);
+    const url = await getS3PutUrl(contentType, contentLength, partialPath);
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/mutations/createOrgPicturePutUrl.js]: heh, yeah, a single test woulda caught this!
i wrote the logic for this thing in the previous PR, but realized the logic was so similar to the user picture mutation that I just held off until this PR to make sure it worked. not being able to test locally is a bummer.

File: src/server/graphql/mutations/login.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+import LoginPayload from 'server/graphql/types/LoginPayload';
+import {
+  auth0AuthenticationClient as auth0Client,
+  clientId as auth0ClientId,
+  clientSecret as auth0ClientSecret
+} from 'server/utils/auth0Helpers';
+import {getUserId} from 'server/utils/authorization';
+import segmentIo from 'server/utils/segmentIo';
+
+const login = {
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/mutations/login.js]: > This just verifies a JWT, and maybe creates an account.

Isn't that a perfect description of what logging is does? :wink:

It was called something like updateUserWithToken, but I could never remember that name. This mutation gets called when you log in, it logs you in, so i figure hey, let's call it log in. No strong opinion on keeping the name.

File: src/server/graphql/rootQuery.js [hunk 12]
```diff
@@ -1,20 +1,16 @@
...
   fields: () => ({
     viewer: {
       type: User,
-      resolve: (source, args, {authToken}) => ({
-        id: authToken.sub
-      })
-    },
-    ...organization,
-    ...teamMember,
-    ...user
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/rootQuery.js]: old format: 1 file, many queries. new format: 1 file has 1 query. They're all down below

File: src/server/graphql/types/CreateUserPicturePutUrlPayload.js [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import {GraphQLObjectType} from 'graphql';
+import GraphQLURLType from 'server/graphql/types/GraphQLURLType';
+
+const CreateUserPicturePutUrlPayload = new GraphQLObjectType({
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/types/CreateUserPicturePutUrlPayload.js]: Hahaha oh man, i hear you! I had some good talks with the GraphQL folks about this about a year ago.

Eventually, they won me over with 2 good arguments:
1. it's more expressive:
```
createOrgPicturePutUrl(args) {
      url
    }

VS.

createOrgPicturePutUrl(args) 
```
In the former, you are asking to get a url back. In the latter, you're asking a scalar result.

2. Future proof
Let's say I want to return 2 things. _now_ my type has changed & all my clients need to update their APIs because it'll be immediately broken (remember, GraphQL schema versioning is an antipattern!)
If I return an object, I can add fields to the payload & everyone can keep on trucking by just ignoring the new field.

File: src/server/graphql/types/NotificationSubscriptionPayload.js [hunk 14]
```diff
@@ -28,6 +29,7 @@ const types = [
   RemoveOrgUserPayload,
   StripeFailPaymentPayload,
   User,
+  UpdateUserProfilePayload,
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/types/NotificationSubscriptionPayload.js]: ohhh do i have a blog post for you :wink: https://medium.com/@matt.krick/the-hybrid-strategy-for-graphql-subscriptions-dd5471c45755

Subscription payloads are unions of mutation payloads, so a project involves notification will live in the e.g. UpdateProjectPayload.

File: src/server/graphql/types/UpdateUserProfilePayload.js [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+const UpdateUserProfilePayload = new GraphQLObjectType({
+  name: 'UpdateUserProfilePayload',
+  fields: () => ({
+    user: {
+      type: User,
+      resolve: resolveUser
+    },
+    teamMembers: {
+      type: new GraphQLList(TeamMember),
+      description: 'The updated team members',
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/types/UpdateUserProfilePayload.js]: oh yes we are! re-read that resolve function :wink:
today, the picture & name is denormalized onto each team member. this was to support subscriptions before graphql supported them.
tomorrow, we could rewrite our front end to something like:
```
teamMember {
  user {
    name
  }
}
```
oh tomorrow....

File: src/universal/components/Auth0ShowLock/signinAndUpdateToken.js [hunk 16]
```diff
@@ -1,16 +1,16 @@
...
-  dispatch(setAuthToken(auth0Token));
-  SendClientSegmentEventMutation(atmosphere, 'User Login');
+  atmosphere.setAuthToken(auth0Token);
+  const onCompleted = (res) => {
+    const {login: {user}} = res;
+    dispatch(setAuthToken(auth0Token, user));
+    SendClientSegmentEventMutation(atmosphere, 'User Login');
+  };
+  const onError = (err) => {
+    console.error('Error logging in', err);
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/universal/components/Auth0ShowLock/signinAndUpdateToken.js]: oh for sure! Now that we're on relay, i can create a default onError handler that calls sentry up with the error message. that's in the chore buffer for me...

File: src/universal/modules/userDashboard/components/OrgAvatarInput/OrgAvatarInput.js [hunk 17]
```diff
@@ -16,18 +16,19 @@ const validate = (values) => {
...
-      reject(err);
-    };
-    const onCompleted = (res) => {
+  const variables = {
+    contentType: pictureFile.type,
+    contentLength: pictureFile.size,
+    orgId
+  };
+  const onError = (err) => {
+    throw new Error(err);
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/universal/modules/userDashboard/components/OrgAvatarInput/OrgAvatarInput.js]: good catch, this should be a `reject` inside the promise...

File: src/universal/modules/userDashboard/components/OrgAvatarInput/OrgAvatarInput.js [hunk 18]
```diff
@@ -16,18 +16,19 @@ const validate = (values) => {
...
+  };
+
+  return new Promise((resolve) => {
+    const onCompleted = async (res) => {
       const {createOrgPicturePutUrl: {url}} = res;
-      const pathname = sendAssetToS3(pictureFile, url);
+      const pathname = await sendAssetToS3(pictureFile, url);
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/universal/modules/userDashboard/components/OrgAvatarInput/OrgAvatarInput.js]: if AWS goes down, we've got much bigger problems than a profile picture :wink:

this is a good point though, we should talk about when to catch `await` clauses. for example, on the server, we always assume our DB is up. if a mutation triggers 2 writes & the 2nd write fails, we don't roll back the first (the exception being around payment stuff. i make sure to avoid race conditions there).

IMO, the effort still isn't worth the reward. if AWS fails, then they'll hit refresh & try again nbd.

File: src/universal/mutations/CreateFirstTeamMutation.js [hunk 19]
```diff
@@ -13,6 +13,9 @@ const mutation = graphql`
         preferredName
       }
       jwt
+      user {
+        ...UserAnalyticsFrag @relay(mask: false)
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/universal/mutations/CreateFirstTeamMutation.js]: all the segment helpers use it to identify the user

File: src/universal/mutations/LoginMutation.js [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+  mutation LoginMutation($auth0Token: String!) {
+    login(auth0Token: $auth0Token) {
+      user {
+        ...UserAnalyticsFrag @relay(mask: false)
+      }
+    }
+  }
+`;
+
+const LoginMutation = (environment, auth0Token, onError, onCompleted) => {
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/universal/mutations/LoginMutation.js]: agreed! oh the patterns that emerge... we should put that on the chores list & do it for all of em.

File: src/universal/mutations/UpdateUserProfileMutation.js (hunk 1 above)
Comment (mattkrick) [reply src/universal/mutations/UpdateUserProfileMutation.js]: 1. it's DRY. mutations & subs want the same payload. why not use a fragment to ensure they get the same payload?
2. we have to mask every fragment on a mutation because Relay lies about their `onCompleted` callback receiving a raw result. if it's masked, then the onCompleted callback will fail.Since I'm not sure which mutations will use onCompleted, i just put it on all of them, because that's a super annoying error.

File: src/universal/mutations/fragments/UserAnalyticsFrag.js (hunk 2 above)
Comment (mattkrick) [reply src/universal/mutations/fragments/UserAnalyticsFrag.js]: yeah, but if not here, where?

File: src/universal/redux/AuthEngine.js (hunk 3 above)
Comment (mattkrick) [reply src/universal/redux/AuthEngine.js]: that's what socket cluster calls it.

File: src/universal/redux/AuthEngine.js (hunk 4 above)
Comment (mattkrick) [reply src/universal/redux/AuthEngine.js]: socket cluster API

File: src/universal/redux/__tests__/AuthEngine.test.js (hunk 5 above)
Comment (mattkrick) [reply src/universal/redux/__tests__/AuthEngine.test.js]: we test it, and we call it,  just not on auth token renewals, because that won't give us anything new.

File: src/universal/modules/userDashboard/containers/UserSettings/UserSettingsContainer.js [hunk 21]
```diff
@@ -38,68 +31,52 @@ const validate = (values) => {
...
-    if (preferredName !== user.preferredName) {
-      this.updateProfile(preferredName)
-        .then(this.onSubmitComplete())
-        .catch((e) => raven.captureException(e));
+    if (preferredName !== viewer.preferredName) {
+      const onError = (err) => {
+        raven.captureException(err);
+        throw new SubmissionError(err);
+      };
+      return new Promise((resolve) => {
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/universal/modules/userDashboard/containers/UserSettings/UserSettingsContainer.js]: fixed!

File: e2e/tests/authentication.test.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (dan-f) [reply e2e/tests/authentication.test.js]: 😬 hehe nothing worse than trying to debug CI... I can give this a look this afternoon!

File: src/server/__tests__/utils/signupTeam.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (dan-f) [reply src/server/__tests__/utils/signupTeam.js]: Aw :( would it be difficult to convert these to using MockDB?

File: src/server/graphql/mutations/createFirstTeam.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (dan-f) [reply src/server/graphql/mutations/createFirstTeam.js]: I see, it's used to generate the `user` field of type `User` by the `resolveUser` function. Yikes, something about that data transformation occurring by way of the `CreateFirstTeamPayload` "type" feels a bit too magical for me. 

Alright, that said - why do we need the `user` field; why do I need the `CreateFirstTeamPayload` to tell me who I am?

File: src/server/graphql/mutations/login.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (dan-f) [reply src/server/graphql/mutations/login.js]: I guess that the "creating an account" part doesn't feel like logging in, but I'm nitpicking here!

File: src/server/graphql/rootQuery.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (dan-f) [reply src/server/graphql/rootQuery.js]: > new format: 1 file has 1 query.

I'm down with this!

> They're all down below

Down below where? All I'm seeing here is that the `Query` type now has a single `viewer` field.

File: src/server/graphql/types/CreateUserPicturePutUrlPayload.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (dan-f) [reply src/server/graphql/types/CreateUserPicturePutUrlPayload.js]: Yeah, the future proof part is what I was thinking too. Cool!

File: src/server/graphql/types/UpdateUserProfilePayload.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (dan-f) [reply src/server/graphql/types/UpdateUserProfilePayload.js]: Alright, if this denormalization doesn't help us anymore, let's get rid of it!

File: src/server/__tests__/utils/signupTeam.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/__tests__/utils/signupTeam.js]: oh we haven't used these tests for about a year. check it out:
```
resolve({}, {newTeam}, {authToken, unitTestCb})
 -        .then((result) => expect(typeof result).toBe('string'))
 -        .catch((e) => console.log(`exception during test: ${e}`));
```

In other words, "when you call this, you get a string back". We don't return strings anymore, and i'd like to know what that string is (granted a little test is better than no test!)

File: src/universal/modules/userDashboard/components/OrgAvatarInput/OrgAvatarInput.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (dan-f) [reply src/universal/modules/userDashboard/components/OrgAvatarInput/OrgAvatarInput.js]: > if AWS goes down, we've got much bigger problems than a profile picture 😉

This could fail for a number of reasons, of which AWS "going down" is just one!

> IMO, the effort still isn't worth the reward. if AWS fails, then they'll hit refresh & try again nbd.

I mean, I hear ya, but I was taught to always catch errors. Even if we just pop a toast saying "try again"...

File: src/universal/mutations/LoginMutation.js (hunk 20 above)
[the remark this answers is not in the document]
Comment (dan-f) [reply src/universal/mutations/LoginMutation.js]: But there's nothing from stopping us from making these changes incrementally, as well!

File: src/universal/mutations/UpdateUserProfileMutation.js (hunk 1 above)
Comment (dan-f) [reply src/universal/mutations/UpdateUserProfileMutation.js]: 1. DRY is good! I'm less concerned here about DRY and more about separation of concerns. Just pointing out that when we share code between two modules, if that shared code isn't biased towards one of it's use cases over the other, then it should probably be in a neutral third location.

File: src/universal/mutations/fragments/UserAnalyticsFrag.js (hunk 2 above)
Comment (dan-f) [reply src/universal/mutations/fragments/UserAnalyticsFrag.js]: I was just pointing gout that it doesn't need to be under `.../mutations/...`. Idk, `universal/fragments`?

File: src/server/graphql/mutations/createFirstTeam.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/mutations/createFirstTeam.js]: Oh we're gonna break you free from those REST shackles yet :wink:
maybe you do want the user, maybe you don't. maybe you didn't but now you do. __the server should't change based on the needs of the client__. Instead, the server says "Hey, createFirstTeam occurred & it affected these people & these things". The client says "Oh yeah? Tell me about that new team". In the future, the client might say "Hey, i also wanna know when the user was created to see if they waited before creating the team". Why should that require a change to the server?

File: src/universal/redux/AuthEngine.js (hunk 3 above)
Comment (dan-f) [reply src/universal/redux/AuthEngine.js]: Sorry that was a crappy comment on my part, haha! I think I was wondering why it's a parameter if it's not being used. Is this class fulfilling an interface for a 3rd party library or something?

File: src/server/graphql/mutations/login.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/mutations/login.js]: yeah, ever since oauth the line between sign up & log in has been really blurred! How many places say "sign up or log in with google". Well, technically, i wanna sign up with you, but through google. On the backend, that don't matter though, because google just gives us a token, they don't tell us if they're a first timer or not, so we don't even know if they're new!

File: src/server/graphql/mutations/createFirstTeam.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (dan-f) [reply src/server/graphql/mutations/createFirstTeam.js]: > Oh we're gonna break you free from those REST shackles yet 😉

??

I get the GraphQL sales pitch. I'm just asking why in this particular instance you want to include the `user` field on this mutation payload. The client is not asking for it (`CreateFirstTeamMutation.js`).

File: src/server/graphql/mutations/login.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (dan-f) [reply src/server/graphql/mutations/login.js]: We know if they're new based on whether they're in our `User` table or not.

File: src/server/graphql/rootQuery.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/rootQuery.js]: They're all under User.js.
So, the person viewing the website is a User type.
That person might want to see what teams they are on in a query like `viewer { teams { name} }`
if we promoted teams to the rootQuery level, then we'd add a `viewerId` arg to the teams query, or it'd be implicit that `teams` only returns teams for that particular viewer.
By following the viewer pattern, anyone reading the query can readily deduce that the teams are derived from the viewer.
Now consider this query:
```
query {
  viewer {
    teams {
      name
    }
  }
  totalParabolUserCount
  helpfulMeetingTips {
    title
  }
}
```

By reading that query, you can assume that it don't matter who you are, you're gonna get the same helpful meeting tips as everyone else.

[the remark this answers is not in the document]
Comment (dan-f) [reply src/server/graphql/rootQuery.js]: So by "down below" you're saying you moved every single query other than `viewer` from being a field on `Query` to being a field on `User`.

File: src/server/graphql/types/UpdateUserProfilePayload.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/types/UpdateUserProfilePayload.js]: i'd do it today if it brought up new users :smile:

File: src/server/graphql/mutations/login.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/mutations/login.js]: the server knows, but the client doesn't.
if we have a login & signup mutation, which one will the client call?

File: src/server/graphql/rootQuery.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (mattkrick) [reply src/server/graphql/rootQuery.js]: ah yeah, down the tree, not down the file.
FWIW this was also borrowed from facebook (and the viewer pattern has some magic built into relay, too)

[the remark this answers is not in the document]
Comment (dan-f) [reply src/server/graphql/rootQuery.js]: It makes sense for Facebook. When you've got a social graph, queries are generally "following your nose" one link at a time.

**Comment by mattkrick** [comment]
E2E tests fixed, see #1662. merging to master
